app/scripts/controllers/onboarding.ts

```diff
@@ -0,0 +1,103 @@
+import { ObservableStore } from '@metamask/obs-store';
+import log from 'loglevel';
+
+type flowTypes = 'create' | 'import';
+/**
+ * The state of the OnboardingController
+ */
+export type OnboardingControllerState = {
+  seedPhraseBackedUp: boolean | null;
+  firstTimeFlowType: flowTypes | null;
+  completedOnboarding: boolean;
+  onboardingTabs?: Record<string, string>;
+};
+
+const defaultTransientState: Record<string, Record<string, string>> = {
+  onboardingTabs: {},
+};
```

Comment:
Same as for `defaultState`, we could use a `satisfies` annotation to a more semantically readable type here.

```suggestion
const defaultTransientState = {
  onboardingTabs: {},
} satisfies Pick<OnboardingControllerState, 'onboardingTabs'>;
```